agent/agent.go

```diff
@@ -228,21 +228,9 @@ func New(c *Config) (*Agent, error) {
 		httpAddrs:       httpAddrs,
 		tokens:          new(token.Store),
 	}
-	if err := a.resolveTmplAddrs(); err != nil {
-		return nil, err
-	}
 
 	// Try to get an advertise address
 	switch {
```

Comment:
tried moving this code over like suggested above, and that's failing a lot of unit test that depends on the test agent having this initialized in here..will be a larger change, but if its worth cleaning up all that i can do it. Don't want to mess with it if @magiconair is going to clean all this up soon anyway,